### DISCUSSION TITLE
Calculating time

### DISCUSSION THREAD

**Discussion by regs01** [Qalculate/libqalculate#371] [General]
Those two approaches gives me different result
`time(3:30−2:15) to time = 20:53:56,25`
`(time(3:30)−time(2:15)) to time = 20:55:31,25`

While those two giving me correct result
`3.5 − 2.25 to time = 1:15`
`3:30 − 2:15 to time = 1:15`

Is this how time function is supposed to work or I misunderstood something or misused it?

**Comment by regs01** [comment]
And more
`3:30 − 2:15 to min = 0,02083333333 min`
even this is same
`(3:30:00.0 − 2:15:00.0) to min = 0,02083333333 min`

Shouldn't result be `75 min`?

**Comment by hanna-kn** [comment]
The time() always function always returns the current time and does not accept any argument.

3:30 is interpreted as a sexagesimal number, which equals 3.5 (and 2:15 equals 2.25). When an expression without any unit is converted to a unit, the initial expression is interpreted as using the corresponding SI base unit (the SI base unit for time is s), thus the expression `3:30 − 2:15 to min` is interpreted as `(3.5 − 2.25) s to min`. I understand that this behaviour is confusing in this case.

**Comment by regs01** [comment]
So it's just this to use `3:30 − 2:15 to time = 1:15` ?

Made it with `03:30 h − 02:15 h to min = 75 min`

**Comment by azizsayfiddin** [comment]
Is there `now` for time?

**Reply by azizsayfiddin** [reply]
I can do this:
```
21:00 - time to time. But, it's verbose.

**Reply by hanna-kn** [reply]
If output format with units is acceptable:
`21h-timeh`
`today+21h-now`

Alternatively the command `base time` can be used before multiple calculations, instead of ` to time` (which however can be slightly shortened to `->time`).
